app/src/main/java/com/example/dungeontest/model/SettingsViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.example.dungeontest.model
+
+import android.app.Application
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.dungeontest.data.SettingsStorage
+import kotlinx.coroutines.launch
+
+class SettingsViewModel( application: Application): AndroidViewModel(application) {
+    private val settingsStorage = SettingsStorage(application)
+    private val _tokenValue = mutableStateOf(TextFieldValue())
+    val tokenValue: State<TextFieldValue> = _tokenValue
+
+    private val _selectedModel = mutableIntStateOf(0)
+    val selectedModel: State<Int> = _selectedModel
```

Comment:
even after going through the whole PR, I'm still finding it confusing that we have State/MutableState objects in the ViewModel. Since you also have MutableState objects in the SettingsScreen composable to keep the ui responsively updating to user input, why not have these viewmodel instance variables be simple strings and ints?